### ISSUE TITLE
[BUG] Spring animation is jumpy when updated

### ISSUE THREAD

**Issue by bakura10** [motiondivision/motion#2963]
Hi,

I am currently upgrading from Motion 10 to 11, and this one is the only one I could not find any workaround for now. I am having a component that follows the cursor and applies a spring animation.

On Motion 10, it is extremely smooth: https://codesandbox.io/p/sandbox/motion-10-spring-5r4gyy

On Motion 11, there is a kind of "delay" that make it feels jumpy. You can reproduce it here: https://codesandbox.io/p/sandbox/motion-11-spring-r5yd3m

I have tried various things, such as using transform instead of x/y/rotate, but on Motion 11 the end result is always extremely jumpy.

Thanks.

**Comment by bakura10** [comment]
After moving to Animation Mini, I can confirm the issue is also reproducible on the mini animate.

**Comment by mattgperry** [comment]
This needs to be throttled with `frame.postRender` since deferred keyframe resolution. Mini version probably suffers the same way because browsers do something similar internally.

Try this: https://codesandbox.io/p/sandbox/motion-11-spring-forked-cxggtg?file=%2Findex.js%3A53%2C60

Nice effect btw

**Comment by bakura10** [comment]
Hi @mattgperry ,

Thanks for hte answer. Could the ticket be re-open ? Unfortunately it does not work for Mini, and the animation is still extremely jumpy. You can try with this:

```
startAnimation = () => {
    animateMini(
      this,
      {
        transform: [
          null,
          `translate(${this.mouseX.toFixed(3)}px, ${(
            this.mouseY -
            350 * (this.translateOffset / 100)
          ).toFixed(3)}px) rotate(${this.rotation}deg)`,
        ],
      },
      {
        type: spring,
        stiffness: 150,
        damping: 15,
      }
    );
  };
```

**Comment by mattgperry** [comment]
Were you using independent transforms in Motion 10 or the transform value itself?

**Comment by bakura10** [comment]
You're correct; I was using independent transforms. I've just tried on Motion 10 with transform, and the result is also not working properly.

So there is no way to achieve this effect smoothly in Animate Mini of Motion 11, right? Isn't there any workaround (maybe using registered properties).

**Comment by mattgperry** [comment]
With the independent values velocity could be calculated, but mini animate doesn’t support these currently. As such there’s no velocity tracking animate mini and so although it could be possible to get spring-ish animations working, it’s definitely not possible to get velocity to feed into the next animation.

**Comment by bakura10** [comment]
Thanks for the confirmation. I should have tested all our use cases before upgrading to Motion 11! Anyway, I think we will need to remove this nice animation.

**Comment by mattgperry** [comment]
Just thrown up an example of the canonical way to do these kinds of spring animations: https://examples.motion.dev/js/spring-follow-cursor
